africanus/model/spi/examples/simple_spi_fitter.py

```diff
@@ -0,0 +1,402 @@
+                  [s(t), c(t)]])
+    A = np.dot(np.dot(R.T, A), R)
+    sOut = xin.shape
+    x = np.array([xin.ravel(), yin.ravel()])
+    R = np.einsum('nb,bc,cn->n', x.T, A, x, optimize=True)
+    fwhm_conv = 2*np.sqrt(2*np.log(2))
+    return np.ascontiguousarray(np.exp(-fwhm_conv*R).reshape(sOut),
+                                dtype=np.float64)
+
+
+def create_parser():
+    p = argparse.ArgumentParser(description='Simple spectral index fitting'
+                                            'tool.',
+                                formatter_class=argparse.RawTextHelpFormatter)
+    p.add_argument("--fitsmodel", type=str, required=True)
+    p.add_argument("--fitsresidual", type=str)
+    p.add_argument('--outfile', type=str,
+                   help="Path to output directory. \n"
+                        "Placed next to input model if outfile not provided.")
+    p.add_argument('--beampars', default=None, nargs='+', type=float,
+                   help="Beam parameters matching FWHM of restoring beam "
+                        "specified as emaj emin pa. \n"
+                        "By default these are taken from the fits header "
+                        "of the residual image.")
+    p.add_argument('--threshold', default=5, type=float,
+                   help="Multiple of the rms in the residual to threshold "
+                        "on. \n"
+                        "Only components above threshold*rms will be fit.")
+    p.add_argument('--maxDR', default=100, type=float,
+                   help="Maximum dynamic range used to determine the "
+                        "threshold above which components need to be fit. \n"
+                        "Only used if residual is not passed in.")
+    p.add_argument('--ncpu', default=0, type=int,
+                   help="Number of threads to use. \n"
+                        "Default of zero means use all threads")
+    p.add_argument('--beammodel', default=None, type=str,
+                   help="Fits beam model to use. \n"
+                        "It is assumed that the pattern is path_to_beam/"
+                        "name_corr_re/im.fits. \n"
+                        "Provide only the path up to name "
+                        "e.g. /home/user/beams/meerkat_lband. \n"
+                        "Patterns mathing corr are determined "
+                        "automatically. \n"
+                        "Only real and imaginary beam models currently "
+                        "supported.")
+    p.add_argument('--output', default='aiIbc', type=str,
+                   help="Outputs to write. Letter correspond to: \n"
+                   "a - alpha map \n"
+                   "i - I0 map \n"
+                   "I - reconstructed cube form alpha and I0 \n"
+                   "b - interpolated beam \n"
+                   "c - restoring beam used for convolution \n"
+                   "Default is to write all of them")
+    return p
+
+
+args = create_parser().parse_args()
```

Comment:
I think it would be safer if the contents of the script were placed in a function lilke so:

```python
def main():
   ....

if __name__ == "__main__":
    main()
```

Briefly, this allows the module to be imported without running the script. https://stackoverflow.com/a/19578357/1611416, which can be useful for tools which inspect python code (such as pytest, probably)